### PR TITLE
Fix IC bugs in ic_proxy_ibuf_push()

### PR DESCRIPTION
**Pull request by interma**
My previous comment 57a314073a59c6ec8ab2b0dfc7ed7f578a4568d3 added magic number field in the ICProxyPkt. Some bugs are found in the process of porting it to 6x: they are related to the invalid value of ICProxyPkt field, current code doesn't handle them since they don't exist in the normal package.

All of them have been fixed in the 6x PR: https://github.com/greenplum-db/gpdb/pull/14998, so this PR backport the fix code and test cases into main branch.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/test/regress/expected/ic_proxy_socket.out [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        icproxy_port = int(addr.split(":")[3])
+except:
+    # not working on icproxy mode, just return
+    icproxy_host = ""
+    pass
+if icproxy_host == "":
+    return
+
+ranstr="fSSAtMbxAzOhkTcpPwvXQq9Y45vfnTiOdo3Mk9V8QKLKWyd8SHBI59FsDRLWFqMZkFpwosqZEIkua3YfR4C8GZ6n6J4SEYCJcdamjhg2e9e1Ns7kNnnUa7OeRyKSEhIANBfau3tsT1SIR4LnXXOAxMBWJkBLtMDKHQyljfa7iQKa1H0bLBP3gOqX8jshoLE7lksNsvmngfpsGtBmKswFNwdLxR1SiMU7kDfU6gCtmLVbCScImf5AVLMosFExTmX3QdiExuuwST6mNZKS7ZdTzsaKbOSy00EWerm39vW8g305H1jrRNSNZvpRkkE2Hv3Abe9ilXZ5hNsWe0DN1i884iPDmxX0B8T8l3SL4P2NInMmcpuQCNVc4ivE7KgLkFafsKrs1igvyRjLraJcjgk88BOkMoVxSE8plK8JoQGDJylg80WFlPBfgm4vjQjRuh8E9MF0M2Ws5GKiDxQ8RcoEEvmHS6ef76kFjai0A10PRkrOPXjNaCJD7jdRPFjkbr3EyhMGNOKMJxXLaUtzuDpz0Ivc0bH9SXffughI61rzChnt71dS15rpTHm6RHodlkwUHfY0H21I3F2uj7qELJR1kb8A5rfYDf0yv8pyDpCeTxT9WjaDlP1DPT74kXZEuIcIfX2H7RvSR89n3W2DJbdgXD9WSz3DknotAzWdY1iNe7qP6hwGJLkxx0ICBD9JaanCwlL4whB1DmtfZqFARsatXPcgVQIYKquJepNoVdVTwSYTb6XBLEGFTn4S6lJXkpyOB9M6egiYIvMYt1HHgDANFmXLslpB3TIYgc3kgGZ2SKI3sjuYghv0TODOLzFZoH9OuLnDkQVXk4a17xpRgl8pal6geryP6sPGA8XHpi0a6gdnqmGiaTQY84EHeykTcgR6K37UVZm9u8Go1xyjqnHjUtcb0BVZg5uY7MUd6ciMAbvXKh837oNpsbkC27qfPawd3W9Bz0aUhxB9st3EksZSmv6b8sRTyEBOT5zwMP7A"
```
Comment (Aegeaner) [inline src/test/regress/expected/ic_proxy_socket.out:26]: Can this `ranstr` be generated on the fly?

Comment (interma) [reply src/test/regress/expected/ic_proxy_socket.out:26]: Thanks Jian, I supposed the static `ranstr` is more stable for test.
After rethinking it, I agree with you, a random string is better: may hit something which I didn't think out before.
I will modify it.

**Review by Aegeaner — APPROVED** [review]
LGTM.

File: src/test/regress/expected/ic_proxy_socket.out (hunk 1 above)
Comment (interma) [reply src/test/regress/expected/ic_proxy_socket.out]: fixed, please review again.

**Pull request merged**